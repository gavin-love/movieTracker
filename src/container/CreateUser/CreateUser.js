import React, { Component } from "react";
import { postNewUser } from "../../utilities/apiCalls/apiCalls";
import { connect } from "react-redux";
import { updateUser, addError, resolveError } from "../../Actions/index";
import "./CreateUser.css";
import PropTypes from 'prop-types'

export class CreateUser extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      email: "",
      password: ""
    };
  }

  handleChange = event => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();
    try {
      const response = await postNewUser(this.state);
      await this.props.handleCreateUser({ id: response.id });
      this.props.clearError();
    } catch (error) {
      this.props.handleError("Email already exists or Password does not match");
    }
  };

  render() {
    return (
      <form className="sign_up" onSubmit={this.handleSubmit}>
        <h5 className="create_title">Create Account</h5>
        <input
          className="create_user"
          type="text"
          name="name"
          value={this.state.name}
          placeholder="name"
          onChange={this.handleChange}
        />
        <input
          className="create_user"
          type="text"
          name="email"
          value={this.state.email}
          placeholder="email"
          onChange={this.handleChange}
        />
        <input
          className="create_user"
          type="password"
          name="password"
          value={this.state.password}
          placeholder="password"
          onChange={this.handleChange}
        />
        <button className="create_button">Submit</button>
      </form>
    );
  }
}
export const mapDispatchToProps = dispatch => ({
  handleCreateUser: user => dispatch(updateUser(user)),
  handleError: error => dispatch(addError(error)),
  clearError: () => dispatch(resolveError())
});

export default connect(
  null,
  mapDispatchToProps
)(CreateUser);

CreateUser.propTypes = {
  handleCreateUser: PropTypes.func,
  handleError: PropTypes.func,
  clearError: PropTypes.func
}