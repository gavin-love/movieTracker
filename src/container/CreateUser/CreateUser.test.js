import React from 'react';
import { shallow } from 'enzyme';
import { CreateUser, mapDispatchToProps } from './CreateUser';

describe('CreateUser', () => {
  let wrapper;
  let mockHandleCreateUser;
  let mockHandleError;

  beforeEach(() => {
    mockHandleCreateUser = jest.fn();
    mockHandleError = jest.fn();
    wrapper = shallow(<CreateUser 
      handleCreateUser={mockHandleCreateUser} 
      handleError={mockHandleError}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have an initial state', () => {
    const expected = {
      name: '',
      email: '', 
      password: ''
    }
    expect(wrapper.state()).toEqual(expected);
  })

describe('handleChange', () => {
  it('should update state', () => {
  let mockEvent = {
    target: {
      name: 'name',
      value: 'cat'
    }
  }
  const expected = {
      name: 'cat',
      email: '', 
      password: ''
    }
  wrapper.instance().handleChange(mockEvent);
  expect(wrapper.state()).toEqual(expected);
})
})
})



