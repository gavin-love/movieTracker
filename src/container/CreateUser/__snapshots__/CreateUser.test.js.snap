// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser
    handleCreateUser={[Function]}
    handleError={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="create_title"
>
          Create Account
</h5>,
        <input
          className="create_user"
          name="name"
          onChange={[Function]}
          placeholder="name"
          type="text"
          value=""
/>,
        <input
          className="create_user"
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="text"
          value=""
/>,
        <input
          className="create_user"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
/>,
        <button
          className="create_button"
>
          Submit
</button>,
      ],
      "className": "sign_up",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create Account",
          "className": "create_title",
        },
        "ref": null,
        "rendered": "Create Account",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "create_user",
          "name": "name",
          "onChange": [Function],
          "placeholder": "name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "create_user",
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "create_user",
          "name": "password",
          "onChange": [Function],
          "placeholder": "password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "create_button",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="create_title"
>
            Create Account
</h5>,
          <input
            className="create_user"
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
/>,
          <input
            className="create_user"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
/>,
          <input
            className="create_user"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            className="create_button"
>
            Submit
</button>,
        ],
        "className": "sign_up",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
            "className": "create_title",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "create_user",
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "create_user",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "create_user",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "create_button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
